### PR TITLE
Fr us 21

### LINKED ISSUE #24: User Story 21, User Can Edit their Password
As a registered user
When I visit my profile page
I see a link to edit my password
When I click on the link to edit my password
I see a form with fields for a new password, and a new password confirmation
When I fill in the same password in both fields
And I submit the form
Then I am returned to my profile page
And I see a flash message telling me that my password is updated

### PR DESCRIPTION
**Pull request by fredrondina**
added views and methods for changing user password from profile

### CODE REVIEW

File: app/views/users/edit.html.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-<%= form_tag '/profile/update', method: :post do |f| %>
+<%= form_tag '/profile/update', method: :patch do |f| %>
```
Comment (kelshadarby) [inline app/views/users/edit.html.erb:2]: Nice catch!

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -50,4 +65,8 @@ def user_params
     )
   end
 
+  def password_params
+    params.permit(:password, :password_confirmation)
+  end
```
Comment (brian-greeson) [inline app/controllers/users_controller.rb:70]: This should be private

**Review by brian-greeson — CHANGES_REQUESTED** [review]
Nice!  just make the params a private method and we are good to go

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -24,6 +24,21 @@ def show
...
+  def update_password
+    user = current_user
+    user.update(password_params)
+    if user.save
+      flash[:notice] = "Password has been changed."
+      redirect_to '/profile'
+    else
+      flash[:error] = "Passwords Must Match"
+      redirect_to '/profile/change_password'
+    end
```
Comment (kelshadarby) [inline app/controllers/users_controller.rb:39]: This method is very easy to read. Nice job!

**Review by kelshadarby — APPROVED** [review]

**Review by brian-greeson — APPROVED** [review]
Looks great!

**Pull request merged**